### DISCUSSION TITLE
Requesting for help with the M0S Dock

### DISCUSSION THREAD

**Discussion by ojaksch** [MiSTle-Dev/MiSTeryNano#30] [Q&A]
According to BLFlashCube I successfully flashed the firmware to the M0S Dock, but when connected to the Tang Nano 20k the M0S gives me only a single LED light. No USB device is working. The USB-C to USB-A adapter is working fine on a normal PC.

I've triple checked the wiring between M0S and the 20k and it is looking error free to me. I could take photos of the wiring if it helps to help me ;)

**Comment by harbaum** [comment]
Have you also flashed the Tang Nano 20k? Is the Atari booting? It should even boot without the M0S.

If it boots, does the RGB led on the Tang Nano 20K light up? If it doesn't the M0S doesn't work. At startup the M0S will tell the TN20K to change the LED color first to blue then to green.

**Comment by ojaksch** [comment]
Yes, the Tang Nano 20k is flashed and working. Atari is booting up and shows our beloved green desktop. Its only the M0S which refuses to work.
When the Tang is booting up, its green LED and the first orange LED  at USB-C lights up. After booting up the Tang's green LED near the HDMI is lighting constantly and the first two little LEDs are flashing alternately.

**Comment by ojaksch** [comment]
Photos of my wiring

![IMG_20240226_142808](https://github.com/harbaum/MiSTeryNano/assets/17717333/0a4e0624-1aa1-4e83-a289-3a47fb09f983)
![IMG_20240226_142851](https://github.com/harbaum/MiSTeryNano/assets/17717333/b6c72e0c-fd7a-48b8-87b5-6797174a5ea1)
![IMG_20240226_142907](https://github.com/harbaum/MiSTeryNano/assets/17717333/2a4e7172-2d50-44c6-99e6-77fd9fd61931)

**Comment by harbaum** [comment]
At a first glance looks good to me. 

Please follow these instructions:

https://github.com/harbaum/MiSTeryNano/blob/main/TROUBLESHOOTING.md

**Comment by ojaksch** [comment]
Na immerhin :)

> If step 2 works and a working M0S is conncted, the the M0S will detect the presence of the FPGA and will instruct it to switch the built-in RGB (ws2812) LED to blue and later to green or red.

Correct, the RGB LED on the Tang turns up blue, then green.
I'll try to find another USB-C -> USB-A adapter... but could it be that are quirky adapters out there?

**Reply by harbaum** [reply]
I'd rather expect a problem between the cherryusb stack and your device.

 What USB device exactly are you trying to use? Did you try a cheap old mouse? These are usually rather uncomplicated.

**Reply by ojaksch** [reply]
I'm using a simple NoName USB-C -> USB-A adapter with a standard USB keyboard. No mouse tested, but I should have tried it.
Also tried a USB-C mini hub with the keyboard but w/o success. Will try two new adapter today - and also with a mouse.

**Reply by ojaksch** [reply]
The two new adapter arrived, but still no joy. Neither keyboard nor mouse is getting powered. No light on device and only a single bright orange LED on M0S.

**Reply by harbaum** [reply]
They don't get powered? There's imho nothing that would prevent 5V from going from the M0S to USB. 

The M0S is powered but the USB is not?

**Reply by ojaksch** [reply]
Have a look at the video. You can see that the USB device (an old simple standard mouse) is getting power for a glimpse, then nothing more.  https://drive.google.com/file/d/1OBcuySJfMfPRF0ycs-XTLhQFS8F2_MlD/view?usp=sharing
I have ordered another M0S just to be safe. Worst case would be a 2nd Nano :)

**Reply by harbaum** [reply]
I would not be too sure that this means the mouse is not getting power. It may simply not be happy with the communication and switch the led off.

Still I have no idea what exactly fails here ... It's actually hard to debug without any debug interface connected to the m0s.

Did you try some different mice?

I actually doubt that the M0S is broken.

**Comment by ojaksch** [comment]
I also have my doubts that it is broken. We'll see today/tomorrow when another M0S arrives.
I'v tested an elderly Logitech Trackball but again without success. While waiting for the new M0S I'll try an arsenal of USB input devices *hehe*

**Comment by ojaksch** [comment]
Oh, and I just used my USB measuring device (with and w/o connected USB device) and it is showing power on the USB pins!

![IMG_20240228_074816](https://github.com/harbaum/MiSTeryNano/assets/17717333/1e18c2fd-e218-4e4a-bb3d-561dad72376e)

**Comment by ojaksch** [comment]
Holy cow! The new M0S arrived some minutes ago, I have the flashed the firmware and - it works!
Now going to troubleshoot the first M0S...

**Reply by harbaum** [reply]
How surprising ...

Hopefully you'll find something. You may set the 'erase entire flash' setting in the cfg.ini I supply with the firmware to see if that changes anything.

**Comment by ojaksch** [comment]
Yeah, I'll try that as the next step before returning it for replacement. Stay tuned.

**Reply by harbaum** [reply]
I am pretty sure it's not physically defective. There's just something that's different and I'd like to understand what. The same may happen to other devices as well.

**Reply by ojaksch** [reply]
Just rewrote the firmware with `erase = 2` in _cfg.ini but still no dice. I could send you that device for testing (I'm located in Germany), but I would lose the warranty claim - which is not the end of the world because of the lost of €8 ;)

**Comment by harbaum** [comment]
Please sent me an email
